liota/dccs/iotcc.py

```diff
@@ -162,21 +162,19 @@ def on_response(msg):
                 log.debug("Processed msg: {0}".format(json_msg["type"]))
                 if json_msg["type"] == "remove_resource_response" and json_msg["body"]["result"] == "succeeded":
                     log.info("Unregistration of resource {0} with IoTCC succeeded".format(entity_obj.ref_entity.name))
+                    if entity_obj.ref_entity.entity_type != "HelixGateway":
+                        self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name,
+                                               entity_obj.ref_entity.entity_type, None, True)
+                    else:
+                        self.remove_reg_entity_details(entity_obj.ref_entity.name, entity_obj.reg_entity_id)
+                        self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name, None, None, True)
                 else:
                     log.info("Unregistration of resource {0} with IoTCC failed".format(entity_obj.ref_entity.name))
             except:
                 raise Exception("Exception while unregistering resource")
 
         self.comms.send(json.dumps(self._unregistration(self.next_id(), entity_obj.ref_entity)))
         on_response(self.recv_msg_queue.get(True, 20))
-        if entity_obj.ref_entity.entity_type != "HelixGateway":
-            self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name,
-                                   entity_obj.ref_entity.entity_type, None, True)
-        else:
-            self.remove_reg_entity_details(entity_obj.ref_entity.name, entity_obj.reg_entity_id)
-            self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name, None, None, True)
-
-        log.info("Unregistration of resource {0} with IoTCC complete".format(entity_obj.ref_entity.name))
```

Comment:
What's the benefit of moving 172-178 to 165? In fact, they are doing the same thing. 
L179 log.info seems useful since L164 just logged unregistration with IoTCC succeeded, while L179 represents the unregistration is fully complete.

Reply:
It is wrongly placed as we figured out in our testing: If the registration of the resource gets failed and enter the else loop at line 165. Then also the below code gets executed and entry is deleted from the json file which shouldn't be the case. Hence, the entry should be deleted from the file only once the un-registration is successfully completed for that resource and your receive the response back.

Line number 179 gets logged even if the resource un-registration is not successful it creates confusion if the resource registration is unsuccessful.

Reply:
I have corrected the un-registration call making it similar to registration call as of now.